ttnn/cpp/ttnn/operations/data_movement/transpose/transpose.hpp

```diff
@@ -6,24 +6,95 @@
 
 #include "ttnn/run_operation.hpp"
 #include "ttnn/decorators.hpp"
-#include "ttnn/deprecated/tt_dnn/op_library/transpose/transpose_op.hpp"
+#include "device/transpose_op.hpp"
+#include "ttnn/operations/data_movement/permute/permute.hpp"
 
 
 namespace ttnn {
 namespace operations::data_movement {
 
 constexpr uint32_t TransposeDefaultQueueID = 0;
 
+inline Tensor transpose_(const Tensor &a, TransposeOpDim transpose_dim, const MemoryConfig& output_mem_config) {
+    bool pad_c = false;
+    bool pad_n = false;
+
+    switch (transpose_dim) {
+        case TransposeOpDim::HC:
+            pad_c = true;
+            break;
+        case TransposeOpDim::NH:
+            return ttnn::permute((const ttnn::Tensor)a, std::vector<int64_t>({2, 1, 0, 3}), output_mem_config);
```

Comment:
why do you need this cast `(const ttnn::Tensor)a` ?

Reply:
I do not, I'll remove